crates/ty_python_semantic/src/types/mro.rs

```diff
@@ -332,8 +332,9 @@ pub(super) enum MroErrorKind<'db> {
     /// This variant records the indices and [`ClassLiteral`]s
     /// of the duplicate bases. The indices are the indices of nodes
     /// in the bases list of the class's [`StmtClassDef`](ruff_python_ast::StmtClassDef) node.
-    /// Each index is the index of a node representing a duplicate base.
-    DuplicateBases(Box<[(usize, ClassLiteral<'db>)]>),
+    /// The first index is the base's first occurrence in the bases list;
+    /// the second index is the base's invalid second occurrence.
```

Comment:
```suggestion
    /// the second index is the base's invalid later occurrence.
```

There can be more than two occurences. Which I think also shows a shortcoming of this implementation because we would still emit three diagnostics for `class Bar(Foo, Foo, Foo, Foo)`. 

Ideally, we would group all repetitive bases together

Reply:
hmm, true

Reply:
Fixed in https://github.com/astral-sh/ruff/pull/17914/commits/1ec9b0ee8588dbe7247177fefd41a8dfa5c9a9c1